### DISCUSSION TITLE
Record on Motion Within Zones?

### DISCUSSION THREAD

**Discussion by DC651F82053AD314** [blakeblackshear/frigate#9238] [Ask A Question] [answered]
I have two cameras setup for object detection through Coral. One of the cameras works great and detects the intended object every time.  The other doesn't always detect the object, so I'm wondering if there is an option to add to the config file to record when motion is detected within a specific zone.  If possible, are there any available examples?
Thanks in advance!

**Comment by NickM-27** [comment]
Motion detection is not that accurate, zones are for objects only. There are many tools to improve object detection results. What issue are you seeing?

**Comment by DC651F82053AD314** [comment]
Thanks for snapping me out of it, Nick!  I agree that motion detection would just cause too many false positives with no convenient object label.  I'm committed to making the object detection better.

I am tracking only one type of object in my config: "Car".  More specifically, they are semi trucks covered in snow.  Now that I've had more action happening in front of the cameras, it's looking like the object detection is working maybe 90% of the time for both cameras/angels.  I'm wondering what I can to improve on that.

Perhaps it's a mix of the angle that the cameras are at, in addition to the snow, that can throw off the object detection?  I wonder if the default Coral AI dataset is having trouble because the dataset may not have enough examples of seeing these types of trucks at this high-up-looking-down angle.  I'm curious whether this is something that I could improve on by using or subscribing to Frigate+ and uploading some of the examples I've captured in Frigate.  What are your thoughts on that?  Are those the tools you're referring to?

Please see the following example screenshots from both cameras to get an idea of what I'm capturing:
Cam1
![image](https://github.com/blakeblackshear/frigate/assets/127766978/6180d9e8-9e92-4860-9b2e-d40298699790)

Cam2
![image](https://github.com/blakeblackshear/frigate/assets/127766978/5a11eb9b-5811-41cd-af8f-94a0a310335c)

I do have a motion mask over the time stamps, and outer areas for each cam.  I'm currently working on configuring the zones, perhaps that will help keep the AI focused on only the truck in the lane, as opposed to also tracking vehicles moving around in the background, I wonder if that will help improve the object identification, as well.

I should note that both the examples above were detected in Frigate. Most do get detected. But every so often it will miss one.

**Reply by NickM-27** [reply, marked as answer]
the default COCO dataset model doesn't have any or at most very few examples from security cameras, so most likely it does not have images from this perspective. You can try lowering the threshold to see if that keeps things working better but top down is a difficult perspective that other users have found the COCO model isn't great at as well.

I would imagine Frigate+ may help as you can tune the model to detect things as you want.

**Reply by DC651F82053AD314** [reply]
Thanks for the input on this.  The semi-trucks are being detected pretty well now.  To build on this original question, I'm now trying to figure out if it would be possible to train the AI to ignore all non semi-trucks.  The only objects I want to be tracked and recorded are the semi-trucks, I'm trying to find a way for it to ignore all other 'cars', as any vehicle that's not a semi-truck is a false positive in my use case.

For example, there's a flatbed truck with a plow that often drives through the lane in front of the cameras and plows the snow.  Is there a way to train the AI to disregard/ignore specific 'cars' by working with Frigate+?
Here is a photo example of what I would want it to ignore:
![image](https://github.com/blakeblackshear/frigate/assets/127766978/28c31bd9-c38f-48bf-9860-08a0b7b9a957)

I'm almost thinking I would want to start with an entirely custom dataset, with only photos of semi-trucks, but I'm not sure if that's possible. Or, I'm wondering if it's possible to dig into the COCO model used in Frigate and edit it, removing all vehicles that aren't semi-trucks, is that possible?

**Reply by NickM-27** [reply]
> Is there a way to train the AI to disregard/ignore specific 'cars' by working with Frigate+?

not currently, no

> I'm almost thinking I would want to start with an entirely custom dataset, with only photos of semi-trucks, but I'm not sure if that's possible. Or, I'm wondering if it's possible to dig into the COCO model used in Frigate and edit it, removing all vehicles that aren't semi-trucks, is that possible?

technically there are separate labels for truck and car but they often switch between each other so frigate just calls them both car. You could try editing the label map so that it is called truck and then only use the truck label and see how it works

**Reply by DC651F82053AD314** [reply]
Thanks for the pointers!  To edit the lable map, I think this is how, is this correct?:
`docker container path: opt/frigate/frigate/labelmap.txt`

enter this in the `labelmap.txt` file:
```
model:
  labelmap:
    8: truck
```
I'm not sure if 'truck' should be where "car" is on line 8 or line 3, but I'm guessing it's line 8, any idea?

**Reply by NickM-27** [reply]
no, there is no reason to edit the labelmap.txt at all. You just add the entry you have into the frigate config file

**Reply by DC651F82053AD314** [reply]
Oh okay, thanks.  If I understand correctly, I can just add the snip below to the config file, change "car" to "truck" everywhere and reboot Frigate.  

```
model:
  labelmap:
    8: truck
```

**Reply by NickM-27** [reply]
yes
